File: implementations/node/core/years/2022/day-10/part-1.mjs
import { Reducer } from "../../../array/reduce.mjs";
import { states } from "./common.mjs";

export function execute(input) {
    let state = states(input)

    let values = []
    for (let i = 19; i < state.length; i += 40) {
        values.push([i + 1, state[i]])
    }

    return values.map(([i, value]) => i * value.x).reduce(Reducer.sum);
}

File: implementations/rust/src/years/y2022/day_10/part_1.rs
use super::common::{calculate_register_history, parse_commands};

pub fn execute(input: String) -> String {
    let commands = parse_commands(input);
    let register_history = calculate_register_history(commands);

    let signal_strength: i32 = register_history[..220].iter()
        .enumerate()
        .skip(19)
        .step_by(40)
        .map(|(i, x)| (i + 1, x))
        .map(|(i, &x)| (i as i32) * x)
        .sum();

    signal_strength.to_string()
}